### PR TITLE
feat(route): Add Kelowna Capital News

### LINKED ISSUE #15520: Kelowna Capital News
### Category

News

### Website URL

https://www.kelownacapnews.com/local-news

### Website description

Kelowna Capital News is a news agency focusing on local news around the Kelowna area.

### What content should be included?

There are several subdirectory, like https://www.kelownacapnews.com/local-news, https://www.kelownacapnews.com/news, https://www.kelownacapnews.com/national-news, https://www.kelownacapnews.com/world-news. 

Since this is a local news agency, the most important category should be https://www.kelownacapnews.com/local-news

### Additional description

_No response_

### This is not a duplicated rss request

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) and [pull requests](https://github.com/DIYgod/RSSHub/pulls) to ensure this rss proposal has not already been requested

### PR DESCRIPTION
**Pull request by hualiong**
## Involved Issue / 该 PR 相关 Issue

Close #15520 

## Example for the Proposed Route(s) / 路由地址示例

```routes
/kelownacapnews/news
/kelownacapnews/local-news
/kelownacapnews/national-news
/kelownacapnews/world-news
/kelownacapnews/entertainment
/kelownacapnews/sports
/kelownacapnews/business
/kelownacapnews/trending-now
/kelownacapnews/opinion
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

Add Kelowna Capital News

### CODE REVIEW

File: lib/routes/kelownacapnews/news.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        const response = await ofetch(`${baseURL}/${type}`);
+        const $ = load(response);
+
+        const list = $('.media')
+            .toArray()
+            .map((item): DataItem => {
+                const a = $(item);
+                return {
+                    title: a.find('.media-heading').text(),
+                    pubDate: timezone(parseDate(a.find('.media-links time').attr('datetime')!), +8),
```
Comment (TonyRL) [inline lib/routes/kelownacapnews/news.ts:52]: The 'Z' in the `datetime` indicates that the time string is in Zulu time, which should not include a timezone offset.

File: lib/routes/kelownacapnews/news.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            });
+
+        const items = await Promise.all(
+            list.map((item) =>
+                cache.tryGet(item.link!, async () => {
+                    const response = await ofetch(item.link!);
+                    const $ = load(response);
+
+                    const details = $('.details-file');
+                    item.image = details.length ? details.find('img').attr('src') : $('#sliderImgs .tablist-item:nth-child(1) img').attr('src');
```
Comment (TonyRL) [inline lib/routes/kelownacapnews/news.ts:64]: `figure` should have their `figcaption` instead of a plain image only

Comment (TonyRL) [inline lib/routes/kelownacapnews/news.ts:64]: Articles with `#sliderImgs` should have all the images with their captions.

File: lib/routes/kelownacapnews/news.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+                    const byline = $('.details-byline');
+                    const profileTitle = byline.find('.profile-title');
+                    if (profileTitle.length) {
+                        item.author = profileTitle.find('a').text();
+                    }
+
+                    item.description = `<img src='${item.image}'></img>${$('.details-body').html()!}`;
```
Comment (TonyRL) [inline lib/routes/kelownacapnews/news.ts:72]: `img` is a self closing element

File: lib/routes/kelownacapnews/news.ts (hunk 2 above)
Comment (hualiong) [reply lib/routes/kelownacapnews/news.ts]: OK，all right...
Wait a few minutes...

File: lib/routes/kelownacapnews/news.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                        item.author = profileTitle.find('a').text();
+                    }
+                    let label = '';
+                    if (typeof image === 'string') {
+                        label = `<img src='${image}' />`;
+                    } else {
+                        for (const e of image!) {
+                            label += `<img src='${$(e).data('src')}' />`;
+                        }
+                    }
```
Comment (TonyRL) [inline lib/routes/kelownacapnews/news.ts:77]: https://github.com/DIYgod/RSSHub/pull/15945#discussion_r1648104815 https://github.com/DIYgod/RSSHub/pull/15945#discussion_r1648106783
captions are still missing

Comment (hualiong) [reply lib/routes/kelownacapnews/news.ts:77]: Uh... You mean the `` tag lacks the `alt` attribute? Or does it mean something else?🤔

Comment (TonyRL) [reply lib/routes/kelownacapnews/news.ts:77]: image
https://developer.mozilla.org/docs/Web/HTML/Element/figcaption

Comment (hualiong) [reply lib/routes/kelownacapnews/news.ts]: OK

File: lib/routes/kelownacapnews/news.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                    const byline = $('.details-byline');
+                    const profileTitle = byline.find('.profile-title');
+                    if (profileTitle.length) {
+                        item.author = profileTitle.find('a').text();
+                    }
+                    let label = '';
+                    if (image.length > 1) {
+                        for (const e of image.toArray()) {
+                            const img = $(e);
+                            label += `<img src='${img.data('src')}' /><figcaption style='margin-bottom: 10px'>${img.attr('title')}</figcaption>`;
```
Comment (TonyRL) [inline lib/routes/kelownacapnews/news.ts:75]: Each `figure` should have one `img` and one `figcaption` only

Comment (hualiong) [reply lib/routes/kelownacapnews/news.ts]: OK

**Pull request merged**